src/Nocarrier/Hal.php

```diff
@@ -234,10 +234,15 @@ public function setData(Array $data = null)
     /**
      * Return an array of data (key => value pairs) representing this resource.
      *
+     * @param null|string data key
      * @return array
      */
-    public function getData()
+    public function getData($key = null)
     {
+        if ($key) {
+            return isset($this->data[$key]) ? $this->data[$key] : null;
```

Comment:
This will break the contract: it should always return an array or an object implementing the Traversable interface.